module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php

```diff
@@ -120,54 +120,130 @@ public function getAccessRestrictions()
+     * specific. Sorted numerically on marc fields.
+     *
+     * @param bool $extended Whether to return a keyed array with the following
+     *  keys:
+     *  - heading: the actual subject heading chunks
+     *  - type: heading type
+     *  - source: source vocabulary
+     *
+     * @return array
+     */
+    protected function getAllSubjectHeadingsNumericalOrder(bool $extended = false): array
+    {
+        $returnValues = [];
         // Try each MARC field one at a time:
         foreach ($this->subjectFields as $field => $fieldType) {
             // Do we have any results for the current field?  If not, try the next.
-            $results = $this->getMarcReader()->getFields($field);
-            if (!$results) {
+            $fields = $this->getMarcReader()->getFields($field);
+            if (!$fields) {
                 continue;
             }
 
             // If we got here, we found results -- let's loop through them.
-            foreach ($results as $result) {
-                // Start an array for holding the chunks of the current heading:
-                $current = [];
-
-                // Get all the chunks and collect them together:
-                foreach ($result['subfields'] as $subfield) {
-                    // Numeric subfields are for control purposes and should not
-                    // be displayed:
-                    if (!is_numeric($subfield['code'])) {
-                        $current[] = $subfield['data'];
-                    }
-                }
-                // If we found at least one chunk, add a heading to our result:
-                if (!empty($current)) {
-                    if ($extended) {
-                        $sourceIndicator = $result['i2'];
-                        $source = $this->subjectSources[$sourceIndicator]
-                            ?? $this->getSubfield($result, '2');
-                        $retval[] = [
-                            'heading' => $current,
-                            'type' => $fieldType,
-                            'source' => $source,
-                            'id' => $this->getSubfield($result, '0'),
-                        ];
-                    } else {
-                        $retval[] = $current;
-                    }
+            foreach ($fields as $f) {
+                if ($nextLine = $this->processSubjectHeadings($f, $extended, $fieldType)) {
+                    $returnValues[] = $nextLine;
                 }
             }
         }
+        return $returnValues;
+    }
 
-        // Remove duplicates and then send back everything we collected:
-        return array_map(
-            'unserialize',
-            array_unique(array_map('serialize', $retval))
-        );
+    /**
+     * Get subject headings of a given record field.
+     * The heading is returned as a chunk, increasing from least specific to most specific.
+     *
+     * @param array  $field        field to handle
```

Comment:
For style reasons, these comments need to be realigned. You can run `composer run fix` from your $VUFIND_HOME directory to automatically correct the problem.